npm.js

```diff
@@ -172,25 +172,55 @@ NPMLocation.configure = function(config, ui) {
   config.remote = config.remote || 'https://npm.jspm.io';
   var npmrc = new Npmrc();
 
+  var npmrcRegistry = npmrc.getRegistry();
+  var npmrcAuth = npmrc.getAuth(npmrcRegistry || defaultRegistry);
+  var hasNpmrc = npmrcAuth || npmrcRegistry;
+
   // check if there are settings in npmrc
-  return rsvp.resolve(config.registry || npmrc.getRegistry() || defaultRegistry)
-  .then(function(resolvedRegistry) {
-    return ui.input('npm registry', resolvedRegistry);
-  })
-  .then(function(registry) {
-    config.registry = registry;
-    if (config.registry.substr(config.registry.length - 1, 1) == '/')
-      config.registry = config.registry.substr(0, config.registry.length - 1);
+  return rsvp.resolve()
```

Comment:
Wow, jam packed namespaces and aliasocity! Didn't realise the tool was at hand, will drop the extra import.